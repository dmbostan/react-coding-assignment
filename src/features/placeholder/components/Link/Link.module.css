.link {
  font-weight: 500;
  color: #646cff;
  text-decoration: inherit;
}

.link:hover {
  color: #535bf2;
}

@media (prefers-color-scheme: light) {
  .link:hover {
    color: #747bff;
  }
}
