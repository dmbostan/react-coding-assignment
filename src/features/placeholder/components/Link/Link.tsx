import { LinkTypes } from './Link.types.ts';
import styles from './Link.module.css';
import classNamesHandler from '@shared/utils/classNamesHandler.ts';

const Link = ({ target, href, className, children }: LinkTypes) => {
  const classNames = classNamesHandler(className, styles.button);

  return (
    <a target={target} href={href} className={classNames}>
      {children}
    </a>
  );
};

export default Link;
