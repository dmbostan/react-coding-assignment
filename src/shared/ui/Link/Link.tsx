import { LinkTypes } from './Link.types.ts';
import styles from './Link.module.css';
import classNamesHandler from '@shared/helpers/classNamesHandler.ts';

const Link = ({ target, href, className, children }: LinkTypes) => {
  const classNames = classNamesHandler(className, styles.link);

  return (
    <a target={target} href={href} className={classNames}>
      {children}
    </a>
  );
};

export default Link;
