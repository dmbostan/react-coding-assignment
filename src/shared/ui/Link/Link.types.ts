import { ReactNode } from 'react';

export type LinkTypes = {
  rel?: string;
  target?: '_blank' | '_self' | '_parent' | '_top';
  href: string;
  className?: string;
  children: string | ReactNode | ReactNode[] | null;
};
