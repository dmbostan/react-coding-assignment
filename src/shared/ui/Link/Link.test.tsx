import Link from './Link.tsx';
import { render, screen } from '@testing-library/react';

describe('Link component', () => {
  it('should render a link with href', () => {
    const url = 'https://example.com';
    const text = 'Example Link';

    render(<Link href={url}>{text}</Link>);

    const linkElement = screen.getByText(text);

    expect(linkElement).toHaveAttribute('href', url);
  });

  it('should render a link with target', () => {
    const url = 'https://example.com';
    const target = '_blank';
    const text = 'Example Link';

    render(
      <Link href={url} target={target}>
        {text}
      </Link>,
    );

    const linkElement = screen.getByText(text);

    expect(linkElement).toHaveAttribute('target', target);
  });

  it('should render a link without target', () => {
    const url = 'https://example.com';
    const text = 'Example Link';

    render(<Link href={url}>{text}</Link>);

    const linkElement = screen.getByText(text);

    expect(linkElement).not.toHaveAttribute('target');
  });

  it('should render a link with class', () => {
    const url = 'https://example.com';
    const className = 'some-class';
    const text = 'Example Text';

    render(
      <Link href={url} className={className}>
        {text}
      </Link>,
    );

    const linkElement = screen.getByText(text);

    expect(linkElement.className).toContain(className);
  });

  it('should match snapshot', () => {
    const url = 'https://example.com';
    const text = 'Example Link';

    const { asFragment } = render(<Link href={url}>{text}</Link>);

    expect(asFragment()).toMatchSnapshot();
  });
});
