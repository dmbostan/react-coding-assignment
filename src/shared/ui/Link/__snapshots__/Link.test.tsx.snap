// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Link component > should match snapshot 1`] = `
<DocumentFragment>
  <a
    class="_link_4e35b2"
    href="https://example.com"
  >
    Example Link
  </a>
</DocumentFragment>
`;
